platforms/aarch64-phytium-pi.toml

```diff
@@ -0,0 +1,80 @@
+# Architecture identifier.
+arch = "aarch64"
+# Platform identifier.
+platform = "aarch64-phytium-pi"
+# Platform family.
+family = "aarch64-phytium-pi"
+
+# Base address of the whole physical memory.
+phys-memory-base = "0x8000_0000"
+# Size of the whole physical memory.
+phys-memory-size = "0x8000_0000" # 2G
+# Base physical address of the kernel image.
+kernel-base-paddr = "0x9000_0000"
+# Base virtual address of the kernel image.
+kernel-base-vaddr = "0xffff_0000_9000_0000"
+# Linear mapping offset, for quick conversions between physical and virtual
+# addresses.
+phys-virt-offset = "0xffff_0000_0000_0000"
+# Kernel address space base.
+kernel-aspace-base = "0xffff_0000_0000_0000"
+# Kernel address space size.
+kernel-aspace-size = "0x0000_ffff_ffff_f000"
+# MMIO regions with format (`base_paddr`, `size`).
+mmio-regions = [
+    ["0x2800_C000", "0x1000"], # UART 0
+    ["0x2800_D000", "0x1000"], # UART 1
+    ["0x2800_E000", "0x1000"], # UART 2
+    ["0x2800_F000", "0x1000"], # UART 3
+
+    ["0x3000_0000", "0x800_0000"],  # other devices
+    ["0x4000_0000", "0x1000_0000"], # Pcie ecam
+
+    ["0x5800_0000", "0x2800_0000"], # 32-bit MMIO space
+
+    ["0x2801_4000", "0x2000"], # MIO0 - I2C
+    ["0x2801_6000", "0x2000"], # MIO1 - I2C
+    ["0x2801_8000", "0x2000"], # MIO2 - I2C
+    ["0x2801_A000", "0x2000"], # MIO3 - I2C
+    ["0x2801_C000", "0x2000"], # MIO4 - I2C
+
+    ["0x000_2803_4000", "0x1000"], # GPIO0 
+    ["0x000_2803_5000", "0x1000"], # GPIO1 
+    ["0x000_2803_6000", "0x1000"], # GPIO2 
+    ["0x000_2803_7000", "0x1000"], # GPIO3 
+    ["0x000_2803_8000", "0x1000"], # GPIO4 
+    ["0x000_2803_9000", "0x1000"], # GPIO5
+]
+
+# UART Address
+uart-paddr = "0x2800_D000"
+# UART Irq num
+uart-irq = "24"
+
+# MIO0 I2C 
+MIO0 = "0x2801_4000"
+
+# PSCI
+psci-method = "smc"
+
+# GICC Address
+gicc-paddr = "0x3080_0000"
+# TODO: gicv3 dosen't support yet, there is no gicd and need a gicr address.
+gicd-paddr = "0x3088_0000"
+
+# Base physical address of the PCIe ECAM space.
+pci-ecam-base = "0x4000_0000"
+# End PCI bus number.
+pci-bus-end = "0x2"
+# PCI device memory ranges.
+pci-ranges = [
+    ["0x0", "0x5000_0000"],               # PIO space
+    ["0x5800_0000", "0x8000_0000"],       # 32-bit MMIO space
+    ["0x10_0000_0000", "0x20_0000_0000"], # 64-bit MMIO space
```

Comment:
The second element of a range is the size, so `0x8000_0000` should be `0x28000000` and `0x20_0000_0000` should be `0x10_0000_0000`.